src/logic/getFields.ts

```diff
@@ -0,0 +1,23 @@
+import { Field, FieldRefs, InternalFieldName } from '../types';
+import { get } from '../utils';
+import isKey from '../utils/isKey';
+import set from '../utils/set';
+
+export default function getFields(
+  fieldsNames: Partial<Record<InternalFieldName, string>>,
+  fieldsRefs: FieldRefs,
+) {
+  const currentFields: Record<InternalFieldName, Field['__field']> = {};
+
+  for (const name in fieldsNames) {
+    const field = get(fieldsRefs, name) as Field | undefined;
+
+    if (field) {
+      !isKey(name)
+        ? set(currentFields, name, field.__field)
+        : (currentFields[name] = field.__field);
+    }
+  }
```

Comment:
```suggestion
    (field.__field) && set(currentFields, name, field.__field);
```

save some bytes? I haven't run the test with this.

Reply:
sorry, I think perhaps we don't need set?

Reply:
I don't know how to save more bytes, here is the output:

```ts
function oe(e, r) {
  const t = {};
  for (const n in e) {
    const e = u(r, n);
    e && (a(n) ? (t[n] = e.__field) : o(t, n, e.__field));
  }
  return t;
}
```

we need `set` for nested fields.

You can merge whether it's good ;)

Reply:
let's get it in. ❤️